### PR TITLE
Fix resource deletion between test classes

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Bugfix

### Description

As part of our refactor, we forget to clear all resources after tests properly, especially in case of test failure. This PR fir this issue. Now, we all resources from previous tests should be properly deleted even during `recreateTestEnv`

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=CustomResourceStatusST,TracingST,ConnectS2I test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=CustomResourceStatusST,TracingST,ConnectS2I test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=TracingST test_only

**Comment by Frawless** [comment]
I tried `TracingST` test many times form my local env and it was green every time. From my POV we should merge it and solve potential issues in new PR (this PR will fix a lot of tests which are failing currently)

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaConnectResource.java [hunk 1]
```diff
@@ -51,17 +52,25 @@ public static DoneableKafkaConnect kafkaConnectWithMetrics(String name, String c
...
             .endMetadata()
-            .withNewSpec()
+            .editOrNewSpec()
                 .withVersion(Environment.ST_KAFKA_VERSION)
-                .withBootstrapServers(KafkaResources.plainBootstrapAddress(name))
+                .withBootstrapServers(KafkaResources.tlsBootstrapAddress(kafkaClusterName))
                 .withReplicas(kafkaConnectReplicas)
+                .withNewTls()
+                    .withTrustedCertificates(new CertSecretSourceBuilder().withNewSecretName(kafkaClusterName + "-cluster-ca-cert").withCertificate("ca.crt").build())
+                .endTls()
```
Comment (ppatierno) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaConnectResource.java]: Why the default Kafka Connect is now TLS?

Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaConnectResource.java]: it's loaded from examples files and default configuration here is with TLS as well

**Pull request merged**